packages/components/src/internal/components/domainproperties/validation/NameExpressionValidationModal.tsx

```diff
@@ -0,0 +1,92 @@
+import React from 'react';
+
+import { ConfirmModal } from '../../../..';
+
+export interface Props {
+    title?: string;
+    warnings: string[];
+    previews: string[];
+    show: boolean;
+    onHide: () => any;
+    onConfirm: () => any;
+}
+
+const nameExpressionWarningPrefix = 'Name Expression warning: ';
+const aliquotNameExpressionWarningPrefix = 'Aliquot Name Expression warning: ';
+
+export class NameExpressionValidationModal extends React.PureComponent<Props> {
+    onConfirm = () => {
+        const { onConfirm, onHide } = this.props;
+        onConfirm();
+        onHide();
+    };
+
+    render() {
+        const { title, onHide, onConfirm, warnings, show, previews } = this.props;
+
+        if (!show || !warnings || warnings.length === 0) return null;
+
+        const nameWarnings = [],
+            aliquotNameWarnings = [];
+        warnings?.forEach(error => {
+            if (error.indexOf(nameExpressionWarningPrefix) === 0)
+                nameWarnings.push(error.substring(nameExpressionWarningPrefix.length));
+            else if (error.indexOf(aliquotNameExpressionWarningPrefix) === 0)
+                aliquotNameWarnings.push(error.substring(aliquotNameExpressionWarningPrefix.length));
+        });
+
+        let warnTitle = title,
+            hasMultiGroup = false;
+        if (!warnTitle) {
+            if (nameWarnings.length > 0 && aliquotNameWarnings.length > 0) {
+                hasMultiGroup = true;
+                warnTitle = 'Sample and Aliquot Naming Expression Warning(s)';
+            } else if (nameWarnings.length > 0) warnTitle = 'Naming Expression Warning(s)';
+            else if (aliquotNameWarnings.length > 0) warnTitle = 'Aliquot Naming Expression Warning(s)';
```

Comment:
Good point. I see that LKB also uses "Pattern", is there a specific place that there is inconsistency?

Reply:
Yes, within the Detail panels for entities within Biologics, the label is "Name Expression" instead of "Naming Pattern".
![Screen Shot 2021-12-22 at 10 51 50 AM](https://user-images.githubusercontent.com/11431178/147141333-2ed007b1-e96d-4519-9447-12b2d53c2372.png)

Reply:
I'm inclined to change the labels in LKB to "Name Expression", because designers already use "Naming Pattern" as label for LKB.

Reply:
I'm confused.  Isn't this what is already done in LKB?  We use "Naming Pattern" consistently in LKSM (or try to), but that's not used in LKB.

Reply:
My bad... meant to say "change the labels in LKB to  "Naming Pattern"", to be consistent with LKSM and also its own designers

Reply:
Maybe file an issue for this rather than wrap that change into this story, unless it's more work to NOT do this now.

Reply:
Sounds good. changes reverted and issue (https://www.labkey.org/home/Developer/issues/issues-details.view?issueId=44551) filed.